core/client/app/components/gh-profile-image.js

```diff
@@ -0,0 +1,76 @@
+/* global md5 */
+import Ember from 'ember';
+
+/**
+ * A component to manage a user profile image. By default it just handles picture uploads,
+ * but if passed a bound 'email' property it will render the user's gravatar image                                                                                   email: '' [description]
+ * @param  {int}     size              The size of the image to render
+ * @param  {Boolean} hasUploadedImage  Whether or not the user has uploaded an image (whether or not to show the default image/gravatar image)
+ * @param  {Boolean} hasEmail          boolean value corresponding to whether the email property has been passed in
```

Comment:
This isn't so much a param as a property? And notably, not a public one so much as a private one (just for templating?)